server/plugin.go

```diff
@@ -98,3 +96,15 @@ func (p *Plugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*mode
 
 	return post, ""
 }
+
+// MessageWillBePosted is invoked when a message is posted by a user before it is committed
+// to the database.
+func (p *Plugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*model.Post, string) {
+	return p.processPost(c, post)
+}
+
+// MessageWillBeUpdated is invoked when a message is updated by a user before it is committed
```

Comment:
Hmm, I'm worried about this change: the autolink plugin frequently breaks my edits, and my only course to recovery is to edit and fix it manually. With this change, it sounds like that path would be taken from me.

Reply:
Could we please keep the discussion _if_ we want this in https://github.com/mattermost/mattermost-plugin-autolink/issues/36? There have been some interesting points there.

Reply:
I get your point. I'm evaluating the posibility of make "updateOnEdit" an option for the pattern. I mean, if you feel comfortable and safe with the pattern, and you know the patter is idempotent, you can enable the updateOnEdit. If not, you can disable it. But is really annoying to not have that possibility.